src/Websites/Websites/Cmdlets/WebApps/PublishAzureWebApp.cs

```diff
@@ -74,6 +74,8 @@ public override void ExecuteCmdlet()
 
             Action zipDeployAction = () =>
             {
+                if (!Path.IsPathRooted(ArchivePath))
+                    ArchivePath = (this.SessionState.Path.CurrentFileSystemLocation).Path + ArchivePath.Split(new string[] { ".." }, StringSplitOptions.None)[1];
```

Comment:
There is a `AsJob` parameter in `Publish-AzWebApp`. If `AsJob` is present, method ExecuteCmdlet() will run as a separate thread. Not sure your code whether works in this scenario. Could you give it a test before I merge your code?

Reply:
Yes, it is working well with Job as well. PFB screenshot FYR. And when I use the `TryResolvePath` method you have suggested, it is failing to execute as Job.

![image](https://user-images.githubusercontent.com/60102891/179905650-6ecdbaee-0efc-42dc-bd7e-746f9d1e2eb1.png)

Reply:
what is the absolute path of 12509issues.zip

Reply:
yes, you are correct. The absolute path here is `D:\SampleApps\12509issue.zip` .

Reply:
Could you try my recommended way and use different locations as test cases?

Reply:
it works only with the `CurrentFileSystemLocation` i.e current directory of the `command prompt ` followed by the given relative path. example:
`D:\>> -ArchivePath '.\SampleApp.Zip\` is equivalent to `D:\SampleApp.Zip`
`D:\>> -ArchivePath '.\Test\SampleApp.Zip\` is equivalent to `D:\Test\SampleApp.Zip`
`D:\>> -ArchivePath '..\Test\SampleApp.Zip\` is equivalent to `D:\Test\SampleApp.Zip`

I have tested all the above scenarios and it is working perfectly.

Reply:
How about going to D:\test\> -ArchivePath '.\SampleApp.Zip?